packages/node_modules/@node-red/runtime/lib/flows/util.js

```diff
@@ -150,488 +215,467 @@ function createNode(flow,config) {
-                    modifiedNodes = modifiedNodes.concat(linkMap[node]);
+
+    var diff = {
+        added:Object.keys(added),
+        changed:Object.keys(changed),
+        removed:Object.keys(removed),
+        rewired:Object.keys(wiringChanged),
+        linked:[],
+        flowChanged: Object.keys(flowChanged),
+        globalConfigChanged
+    }
+
+    // Traverse the links of all modified nodes to mark the connected nodes
+    var modifiedNodes = diff.added.concat(diff.changed).concat(diff.removed).concat(diff.rewired);
+    var visited = {};
+    while (modifiedNodes.length > 0) {
+        node = modifiedNodes.pop();
+        if (!visited[node]) {
+            visited[node] = true;
+            if (linkMap[node]) {
+                if (!changed[node] && !added[node] && !removed[node] && !wiringChanged[node]) {
+                    diff.linked.push(node);
                 }
+                modifiedNodes = modifiedNodes.concat(linkMap[node]);
             }
         }
-        // console.log(diff);
-        // for (id in newConfig.allNodes) {
-        //     console.log(
-        //         (added[id]?"a":(changed[id]?"c":" "))+(wiringChanged[id]?"w":" ")+(diff.linked.indexOf(id)!==-1?"l":" "),
-        //         newConfig.allNodes[id].type.padEnd(10),
-        //         id.padEnd(16),
-        //         (newConfig.allNodes[id].z||"").padEnd(16),
-        //         newConfig.allNodes[id].name||newConfig.allNodes[id].label||""
-        //     );
-        // }
-        // for (id in removed) {
-        //     console.log(
-        //         "- "+(diff.linked.indexOf(id)!==-1?"~":" "),
-        //         id,
-        //         oldConfig.allNodes[id].type,
-        //         oldConfig.allNodes[id].name||oldConfig.allNodes[id].label||""
-        //     );
-        // }
-
-        return diff;
-    },
-
-    /**
-     * Create a new instance of a node
-     * @param  {Flow} flow     The containing flow
-     * @param  {object} config The node configuration object
-     * @return {Node}          The instance of the node
-     */
-    createNode: createNode
+    }
+    // console.log(diff);
+    // for (id in newConfig.allNodes) {
+    //     if (added[id] || changed[id] || wiringChanged[id] || diff.linked.indexOf(id)!==-1) {
+    //         console.log(
+    //             (added[id]?"a":(changed[id]?"c":" "))+(wiringChanged[id]?"w":" ")+(diff.linked.indexOf(id)!==-1?"l":" "),
+    //             newConfig.allNodes[id].type.padEnd(10),
+    //             id.padEnd(16),
+    //             (newConfig.allNodes[id].z||"").padEnd(16),
+    //             newConfig.allNodes[id].name||newConfig.allNodes[id].label||""
+    //         );
+    //     }
+    // }
+    // for (id in removed) {
+    //     console.log(
+    //         "- "+(diff.linked.indexOf(id)!==-1?"~":" "),
+    //         id,
+    //         oldConfig.allNodes[id].type,
+    //         oldConfig.allNodes[id].name||oldConfig.allNodes[id].label||""
+    //     );
+    // }
```

Comment:
```suggestion
```

Reply:
No - leaving this code in place because it is super useful to be able to uncomment when debugging anything to do with the code. I know its in the git history and ordinarily I'd be all for removing, but from experience, future me will be grateful it's been kept in.